[css-anchor-position-1] Import Web Platform Tests

RWDavid opened the pull request:
#### 452498e430435e95d91db10233e832c0f20db655
<pre>
[css-anchor-position-1] Import Web Platform Tests
<a href="https://bugs.webkit.org/show_bug.cgi?id=274926">https://bugs.webkit.org/show_bug.cgi?id=274926</a>
<a href="https://rdar.apple.com/129025250">rdar://129025250</a>

Reviewed by Ryan Reno, Matthieu Dubet and Sammy Gill.

Import CSS anchor positioning WPT.

Upstream commit: <a href="https://github.com/web-platform-tests/wpt/commit/c426a4354c319e31dd3efc39652b1192603aab75">https://github.com/web-platform-tests/wpt/commit/c426a4354c319e31dd3efc39652b1192603aab75</a>

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/resources/import-expectations.json:
* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/

Canonical link: <a href="https://commits.webkit.org/279590@main">https://commits.webkit.org/279590@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/2bd808c3688cdce96c0a4ed5c4c3a3c4af513a53

| Misc | iOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/53881 "Passed style check") | [  ~~🛠 ios~~](https://ews-build.webkit.org/#/builders/48/builds/33250 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 mac~~](https://ews-build.webkit.org/#/builders/55/builds/6405 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 wpe~~](https://ews-build.webkit.org/#/builders/5/builds/57160 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 wincairo~~](https://ews-build.webkit.org/#/builders/59/builds/4604 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") 
| | [  ~~🛠 ios-sim~~](https://ews-build.webkit.org/#/builders/49/builds/40752 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 mac-AS-debug~~](https://ews-build.webkit.org/#/builders/61/builds/4479 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 wpe-wk2~~](https://ews-build.webkit.org/#/builders/5/builds/57160 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 wincairo-tests~~](https://ews-build.webkit.org/#/builders/59/builds/4604 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") 
| [  ~~🧪 webkitperl~~](https://ews-build.webkit.org/#/builders/11/builds/55979 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/49/builds/40752 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 api-mac~~](https://ews-build.webkit.org/#/builders/55/builds/6405 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 api-wpe~~](https://ews-build.webkit.org/#/builders/5/builds/57160 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| | [  ~~🧪 ios-wk2-wpt~~](https://ews-build.webkit.org/#/builders/49/builds/40752 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-wk1~~](https://ews-build.webkit.org/#/builders/55/builds/6405 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 wpe-cairo~~](https://ews-build.webkit.org/#/builders/65/builds/2759 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/49/builds/40752 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-wk2~~](https://ews-build.webkit.org/#/builders/55/builds/6405 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🛠 gtk~~](https://ews-build.webkit.org/#/builders/2/builds/58754 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| | [  ~~🛠 tv~~](https://ews-build.webkit.org/#/builders/44/builds/29054 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/61/builds/4479 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/2/builds/58754 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/46/builds/30247 "Built successfully") | [  ~~🧪 mac-wk2-stress~~](https://ews-build.webkit.org/#/builders/55/builds/6405 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | [  ~~🧪 api-gtk~~](https://ews-build.webkit.org/#/builders/2/builds/58754 "EWS skipped this build as PR had skip-ews label when EWS attempted to process it.") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/11738 "Built successfully and passed tests") | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/43/builds/31193 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/45/builds/30024 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

stwrt commented on LayoutTests/TestExpectations:
```diff
@@ -7501,3 +7501,71 @@ webkit.org/b/272417 imported/w3c/web-platform-tests/svg/path/property/priority.s
 webkit.org/b/271692 imported/w3c/web-platform-tests/css/css-align/blocks/align-content-block-break-overflow-020.html [ ImageOnlyFailure ]
+
+# Anchor positioning not yet implemented
+webkit.org/b/274926 imported/w3c/web-platform-tests/css/css-anchor-position/anchor-center-002.html [ ImageOnlyFailure ]
```
Does it make sense to mark these as `Skip`, since they are not implemented?

RWDavid replied:
Makes sense. I will change them.

sammygill replied:
I don't think we should be skipping these tests just because we don't implement the feature. We lose out on information when they start passing as you work on the feature and I don't think they are expensive enough to justify skipping.

stwrt reviewed (commented):
Overall it looks good to me. Left one comment.

You probably do not need to include every imported test case in the commit message. Just listing the overarching folder should be fine here as the commit message is now hundreds of lines long.

rreno reviewed (approved):
r=me assuming green bots.

mdubet commented on LayoutTests/TestExpectations:
```diff
@@ -7501,3 +7501,71 @@ webkit.org/b/272417 imported/w3c/web-platform-tests/svg/path/property/priority.s
 webkit.org/b/271692 imported/w3c/web-platform-tests/css/css-align/blocks/align-content-block-break-overflow-020.html [ ImageOnlyFailure ]
+
+# Anchor positioning not yet implemented
+webkit.org/b/274926 imported/w3c/web-platform-tests/css/css-anchor-position/anchor-center-002.html [ Skip ]
```
For future knowledge, we can `Skip` whole folder instead of file by file
`webkit.org/b/274926 imported/w3c/web-platform-tests/css/css-anchor-position/ [ Skip ]`

But in this PR, we should not `Skip` them

sammygill commented on LayoutTests/TestExpectations:
```diff
@@ -7501,3 +7501,90 @@ webkit.org/b/272417 imported/w3c/web-platform-tests/svg/path/property/priority.s
+# Anchor positioning is not yet implemented...
+
+# general failures
+webkit.org/b/274926 imported/w3c/web-platform-tests/css/css-anchor-position/anchor-center-002.html [ ImageOnlyFailure ]
```
These links are usually open bugzillas that correspond to the underlying bug and get resolved when the bug is fixed. Since this bugzilla, which just reflects the import of these tests, will be resolved by the time this patch lands we should not put it here. You can probably just not include any link at all

RWDavid replied:
Okay, will remove the links.

RWDavid committed:
[css-anchor-position-1] Import Web Platform Tests
https://bugs.webkit.org/show_bug.cgi?id=274926
rdar://129025250

Reviewed by Ryan Reno, Matthieu Dubet and Sammy Gill.

Import CSS anchor positioning WPT.

Upstream commit: https://github.com/web-platform-tests/wpt/commit/c426a4354c319e31dd3efc39652b1192603aab75

* LayoutTests/TestExpectations:
* LayoutTests/imported/w3c/resources/import-expectations.json:
* LayoutTests/imported/w3c/web-platform-tests/css/css-anchor-position/

Canonical link: https://commits.webkit.org/279590@main
